File: BE/Booking-Hotel_BE/Booking-Hotel/Models/Promotion.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Booking_Hotel.Models
{
    public class Promotion
    {
        public Promotion() {
            Status = "Active";
        }
        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        public string Title { get; set; }

        [Required]
        public double Discount { get; set; }

        [StringLength(50)]
        public string CouponCode { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [StringLength(999)]
        public string? Description { get; set; }

        [StringLength(50)]
        public string Status { get; set; }

        public int HotelID { get; set; }

        [ForeignKey("HotelID")]
        public Hotel? Hotel { get; set; }
    }
}

File: FE/Booking-Hotel_user_FE/src/app/pay/promotion.ts
import {User} from "../user/user";

export interface Promotion {
  Id: number;
  Title: string;
  Discount: number;
  CouponCode: string;
  StartDate: Date;
  EndDate: Date;
  Description: string;
  Status: string;
  HotelID: number;
  HotelName?: string;
  couponCode : string;
  discount : number;
  title: string;
  endDate: Date;
  startDate: Date;
  hotelID: number;
}

export interface Payment {
  id: number;
  userId: string;
  description: string;
  status: string;
  name: string;
  user: User;
}
